### DISCUSSION TITLE
Are there build options for only Telegram, only WhatsApp?

### DISCUSSION THREAD

**Discussion by jianguo928** [d99kris/nchat#108] [Q&A]
Thanks for you sharing. It is a awesome project.
Could you please provide a option for the user who only need whatsapp or telegram?
Best regards.

**Comment by d99kris** [comment]
Thanks! Actually this is already supported, just not yet properly documented.

The protocols supported by nchat is controlled by the following cmake flags:

    HAS_DUMMY=ON
    HAS_TELEGRAM=ON
    HAS_WHATSAPP=ON

It is possible to enable / disable protocols by passing one or multiple flags to cmake:

    mkdir -p build && cd build
    cmake -DHAS_WHATSAPP=OFF .. && make -s

I will create an issue from this discussion for documenting this in the README.

**Reply by d99kris** [reply]
The README has been updated now.

**Comment by gyungchaehan** [comment]
> Thanks! Actually this is already supported, just not yet properly documented.
> 
> The protocols supported by nchat is controlled by the following cmake flags:
> 
> ```
> HAS_DUMMY=ON
> HAS_TELEGRAM=ON
> HAS_WHATSAPP=ON
> ```
> 
> It is possible to enable / disable protocols by passing one or multiple flags to cmake:
> 
> ```
> mkdir -p build && cd build
> cmake -DHAS_WHATSAPP=OFF .. && make -s
> ```
> 
> I will create an issue from this discussion for documenting this in the README.

Can this be done with the AUR version of nchat on Arch?

**Reply by d99kris** [reply]
Hi @gyungchaehan - unfortunately I don't think it is. I'm not 100% sure though (I do not maintain the AUR packaging).

Perhaps you can try contact the AUR package maintainers of https://aur.archlinux.org/packages/nchat or https://aur.archlinux.org/packages/nchat-git and see if they have any solution.

Otherwise it's always possible to clone this repository and build it from source and specifying the `HAS_` options mentioned in a previous comment.
